DependencyInjection/Configuration.php

```diff
@@ -108,4 +109,95 @@ protected function configureClients(ArrayNodeDefinition $root)
                 ->end()
             ->end();
     }
+
+    /**
+     * @param ArrayNodeDefinition $root
+     */
+    protected function configurePlugins(ArrayNodeDefinition $root)
+    {
+        $root->children()
+            ->arrayNode('plugins')
+                ->addDefaultsIfNotSet()
+                ->children()
+
+                    ->arrayNode('authentication')
+                    ->canBeEnabled()
+                        ->children()
+                            ->scalarNode('authentication')->isRequired()->cannotBeEmpty()->end()
+                        ->end()
+                    ->end() // End authentication plugin
+
+                    ->arrayNode('cache')
+                    ->canBeEnabled()
+                    ->addDefaultsIfNotSet()
+                        ->children()
+                            ->scalarNode('cache_pool')->isRequired()->cannotBeEmpty()->end()
+                            ->scalarNode('stream_factory')->cannotBeEmpty()->defaultValue('httplug.stream_factory')->end()
+                            ->arrayNode('config')
+                                ->children()
+                                    ->scalarNode('default_ttl')->defaultNull()->end()
+                                    ->scalarNode('respect_cache_headers')->defaultTrue()->end()
+                                ->end()
+                            ->end()
+                        ->end()
+                    ->end() // End cache plugin
+
+                    ->arrayNode('cookie')
+                    ->canBeEnabled()
+                        ->children()
+                            ->scalarNode('cookie_jar')->isRequired()->cannotBeEmpty()->end()
+                        ->end()
+                    ->end() // End cookie plugin
+
+                    ->arrayNode('decoder')
+                    ->canBeDisabled()
+                    ->addDefaultsIfNotSet()
+                        ->children()
+                            ->scalarNode('use_content_encoding')->defaultTrue()->end()
+                        ->end()
+                    ->end() // End decoder plugin
+
+                    ->arrayNode('history')
+                    ->canBeEnabled()
+                        ->children()
+                            ->scalarNode('journal')->isRequired()->cannotBeEmpty()->end()
```

Comment:
the debug toolbar history is auto-injected into every client we know of, when we determine that the toolbar is wanted. what is the use for this plugin? i would have to add it manually, right? i don't see the use of having it in the configuration like this, and would prefer dropping it and explaining in the documentation why its not in there.

i think this would just clutter the configuration with things that won't be used much, if at all.